ee/tabby-webserver/src/service/background_job/mod.rs

```diff
@@ -36,6 +37,43 @@ pub enum BackgroundJobEvent {
     SerializedBackgroundJob(String),
 }
 
+impl BackgroundJobEvent {
+    fn job_name(&self) -> Option<&'static str> {
+        match self {
+            BackgroundJobEvent::SchedulerGitRepository(_) => Some(SchedulerGitJob::NAME),
+            BackgroundJobEvent::SchedulerGithubGitlabRepository(_) => {
+                Some(SchedulerGithubGitlabJob::NAME)
+            }
+            BackgroundJobEvent::SyncThirdPartyRepositories(_) => Some(SyncIntegrationJob::NAME),
+            BackgroundJobEvent::WebCrawler(_) => Some(WebCrawlerJob::NAME),
+            BackgroundJobEvent::SerializedBackgroundJob(s) => {
+                let event = serde_json::from_str::<Self>(s);
+                event.ok().and_then(|event| event.job_name())
+            }
+        }
+    }
+
+    fn string_param(&self) -> Option<String> {
+        match self {
+            BackgroundJobEvent::SchedulerGitRepository(config) => Some(config.git_url.clone()),
+            BackgroundJobEvent::SchedulerGithubGitlabRepository(id) => Some(id.to_string()),
+            BackgroundJobEvent::SyncThirdPartyRepositories(id) => Some(id.to_string()),
+            BackgroundJobEvent::WebCrawler(url) => Some(url.clone()),
```

Comment:
Please use serde_json::to_string - as we don't really have a way to control underlying parameter change. Imagine what if some other engineer adds a field to `RepositoryConfig` and affects the behavior of job?